src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java

```diff
@@ -29,18 +31,28 @@ public class FinancialTransactionServiceImpl implements FinancialTransactionServ
     private final FinancialTransactionRepository financialTransactionRepository;
     private final FinancialTransactionModelMapper financialTransactionModelMapper;
     private final WalletRepository walletRepository;
+    private final FinancialTransactionCategoryRepository financialTransactionCategoryRepository;
     @Override
     public FinancialTransactionDTO createFinancialTransaction(@Valid FinancialTransactionCreateDTO financialTransactionCreateDTO) {
         Long walletId = financialTransactionCreateDTO.walletId();
         Wallet wallet = walletRepository.findById(walletId).orElseThrow(() -> {
             throw new AppRuntimeException(ErrorCode.W003, String.format("Wallet with id: %d does not exist", walletId));
         });
+        FinancialTransactionCategory financialTransactionCategory = null;
+        Long categoryId = financialTransactionCreateDTO.categoryId();
+
+        if(!(financialTransactionCreateDTO.categoryId() == null)
```

Comment:
```suggestion
        if (categoryId != null
```

Reply:
If the category with the given id does not exist, throw an exception FTC001.

Reply:
Że jeżeli w wyniku porównania dwóch obiektów przy pomocy metody equals() otrzymamy wartość true (gdy obiekty są sobie równe), to wynik metody hashCode() dla tych obiektów powinien być identyczny.” Ale „kontrakt ten wcale nie musi być spełniony w drugą stronę, czyli jeżeli dla dwóch obiektów metoda hashCode() zwraca taką samą wartość, to wcale nie oznacza to, że obiekty te są sobie równe. Także proszę o porówne, !equals()